packages/db/postgres/functions.sql

```diff
@@ -214,6 +214,44 @@ BEGIN
 END
 $$;
 
+CREATE OR REPLACE FUNCTION upsert_pins(data json) RETURNS VOID
+    LANGUAGE plpgsql
+    volatile
+    PARALLEL UNSAFE
+AS
+$$
+DECLARE
+  pin_data json;
+  pin_location_result_id BIGINT;
+  pin_result_id BIGINT;
+BEGIN
+  FOREACH pin_data IN array json_arr_to_json_element_array(data -> 'pins')
+  LOOP
+    -- Add to pin_location table if new
```

Comment:
Could we be calling `upsert_pin` here instead? 
We'd avoid some duplication. There's an inconsistency with the data inputs, in one function we use `content_cid` and in the other we use`cid`, but I think that can be easily changed? And we'd benefit from a bit more consistency?

Reply:
Corrected to re-use `upsert_pin` and to return an array of pin Ids.
Also changed to use content_cid consistenly.